helpers/embeds.js

```diff
@@ -0,0 +1,65 @@
+import { html, nothing, render } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { map } from 'lit/directives/map.js';
+import { tryGetPluginByKey } from './plugins.js';
+
+const embedTypeAttributeName = 'data-d2l-embed-type';
+
+export async function createEmbedPlaceholder(embedType, props) {
+	const embedRendererPlugin = tryGetPluginByKey('d2l-html-embed-renderer', embedType);
+	if (!embedRendererPlugin) return;
+
+	const placeholderData = await embedRendererPlugin.getPlaceholder(props);
+	const contents = placeholderData.contents
+		? map(Object.entries(placeholderData.contents), ([id, content]) => html`<template data-d2l-embed-template-id=${id}>${content}</template>`)
+		: nothing;
+
+	const attributes = placeholderData.attributes ? JSON.stringify(placeholderData.attributes) : undefined;
```

Comment:
Nit: I think we should call this `properties` instead of `attributes`, since they're more inline with the concept of camelCase strongly-typed "properties" vs. kebab-cased everything-is-a-string "attributes".

Reply:
Good thought. I can change the verbiage here.